core/cas-server-core-monitor/src/main/java/org/apereo/cas/monitor/SystemMonitorHealthIndicator.java

```diff
@@ -0,0 +1,54 @@
+package org.apereo.cas.monitor;
+
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.boot.actuate.metrics.MetricsEndpoint;
+
+import java.util.List;
+
+/**
+ * Monitors JVM system load and memory.
+ *
+ * @author Travis Schmidt
+ * @since 6.1.0
+ */
+@RequiredArgsConstructor
+public class SystemMonitorHealthIndicator extends AbstractHealthIndicator {
+
+    private final MetricsEndpoint metrics;
+    private final int threshold;
+
+    @Override
+    protected void doHealthCheck(final Health.Builder builder) {
+        val systemUsage = metrics.metric("system.cpu.usage", null).getMeasurements().get(0).getValue();
```

Comment:
Refactor into a smaller method with such that `getMeMetricsFor(metrics, "system.cpu.usage")`, etc.